From a77191b0ecad5066e386b3512946b577ac61a078 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Tue, 7 Jan 2025 11:33:41 +0100
Subject: [PATCH 1434/1695] dts: stm32: keep vddio2 and vddio3 always-on on
 stm32mp215f-dk

Keep vddio2 and vddio3 as they are not handled by non secure.
vddio3 is used by the following peripherals:
*SPI1*
PD1 SPI1_MISO
PD2 SPI1_NSS
PD11 SPI1_SCK

*CAMERA*
PD0 CAM_ENABLE
PD5 CAM_RESET

*SDMMC*
PD3 SD.uSD_DETECT

*SAI1*
PD4 SAI1_SDB
PD7 SAI1_FSA
PD8 SAI1_SDA
PD9 SAI1_SCKA

*BLUETOOTH*
PD6 BT_HOST_WAKE

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e6c131b649311ba511146876aa82202820244111 dts: stm32: keep vddio2 and vddio3 always-on on stm32mp215f-dk]
Change-Id: I85a7654f354e034274d908b56faf4ea567166084
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/427218
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp215f-dk.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index 29f5936e9..abe9667f1 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -299,11 +299,15 @@
 	vddio2 {
 		status = "okay";
 		vddio2-supply = <&vdda_1v8>;
+		/* Keep always-on because it can not be enabled by non secure */
+		regulator-always-on;
 	};
 
 	vddio3 {
 		status = "okay";
 		vddio3-supply = <&vddio_pmic>;
+		/* Keep always-on because it can not be enabled by non secure */
+		regulator-always-on;
 	};
 
 	vddio {
-- 
2.47.2

